### PR TITLE
Ensure the required Spree engines are included before decorating classes.

### PR DESCRIPTION
**Pull request by kgleong**
* Excluding the `Spree::Frontend::Engine` from an application results in errors.
* Monkeypatch only when then engine containing the patched class has been included.

### CODE REVIEW

**Comment by kgleong** [comment]
will write specs if approach is approved

**Comment by SebAshton** [comment]
There is a lot of set up going on inside `SpreeMultiDomain::Config` which might be better suited to a service object?

File: app/controllers/spree/api/line_items_controller_decorator.rb [hunk 1]
```diff
@@ -1 +1,5 @@
-Spree::Api::LineItemsController.include SpreeMultiDomain::CreateLineItemSupport
+unless SpreeMultiDomain::Config.ignore_api_controllers?
+  Spree::Api::LineItemsController.include(
+    SpreeMultiDomain::CreateLineItemSupport
```
Comment (SebAshton) [inline app/controllers/spree/api/line_items_controller_decorator.rb]: Any reason why these includes have been put on to new lines? I think it makes the code less readable. Same for the others.

Comment (kgleong) [inline app/controllers/spree/api/line_items_controller_decorator.rb]: no problem i can revert those back to single line

**Comment by kgleong** [comment]
@SebAshton thanks for taking a look.  The login in the `Config` module is fairly simple, and I followed the rails docs on configuring engines here: http://guides.rubyonrails.org/engines.html#configuring-an-engine

The extra methods are trivial helper methods which I'm not sure warrant an extra layer.  I've also added tests

**Comment by adammathys** [comment]
We definitely need to be more conscious of this type of scenario. However, I don't think additional configuration is the best approach. I would rather adopt a [similar approach](https://github.com/solidusio/solidus_auth_devise/blob/master/lib/spree/auth/engine.rb#L78-L88) to `solidus_auth_devise`. Which will allow us to safely include our extensions to those engines without having to force any additional configuration.

What do you think, @kgleong?

**Comment by kgleong** [comment]
@adammathys I like that approach.  Less hassle for the client.  I'll refactor.

**Comment by adammathys** [comment]
:+1: Thanks @kgleong!

**Pull request merged**